pkg/controller/servicebinding/secret.go

```diff
@@ -27,17 +33,90 @@ func (s *secret) buildResourceClient() dynamic.ResourceInterface {
 	return s.client.Resource(gvr).Namespace(s.ns)
 }
 
-// createOrUpdate will take informed payload and either create a new secret or update an existing
-// one. It can return error when Kubernetes client does.
-func (s *secret) createOrUpdate(payload map[string][]byte, ownerReference metav1.OwnerReference) (*unstructured.Unstructured, error) {
+// check if payload secret is same as SBR Secret
+func (s *secret) isSame(SBRSecret string) bool {
 	logger := s.logger.WithValues("Namespace", s.ns, "Name", s.name)
+
+	// fetch SBR secret
+	resourceClient := s.buildResourceClient()
+	existingSecret, err := resourceClient.Get(
+		SBRSecret,
+		metav1.GetOptions{},
+	)
+	if err != nil {
+		logger.Error(err, "Error fetching secret")
+		return false
+	}
+	existingSecretData, _, _ := unstructured.NestedMap(existingSecret.Object, "data")
+
+	// fetch new payload secret
+	payloadInterim := make(map[string]interface{})
+	for k, v := range s.data {
+		payloadInterim[k] = base64.StdEncoding.EncodeToString(v)
+	}
+
+	comparisonResult := nestedMapComparison(existingSecretData, payloadInterim)
+
+	if comparisonResult.Success {
+		logger.Debug("Secret is same.")
+		return true
+	}
+
+	return false
+}
+
+// secretDataHash creates hash of secret data using sha1
+func secretDataHash(data map[string][]byte) (string, error) {
+	keys := make([]string, 0, len(data))
+	for k := range data {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	hash := sha1.New()
+	for _, k := range keys {
+		_, err := hash.Write([]byte(k))
+		if err != nil {
+			return "", err
+		}
+		_, err = hash.Write(data[k])
+		if err != nil {
+			return "", err
+		}
+	}
+	return hex.EncodeToString(hash.Sum(nil)), nil
+}
+
+func (s *secret) Name() (string, error) {
+	hash, err := secretDataHash(s.data)
+	if err != nil {
+		s.logger.Error(err, "Error creating unique secret name hash")
+		return "", err
+	}
+	if strings.Contains(s.name, hash) {
+		return s.name, nil
+	}
+	return s.name + "-" + hash, nil
+}
+
+// create will take informed payload and create a new secret
+// It can return error when Kubernetes client does.
+func (s *secret) create() (*unstructured.Unstructured, error) {
+	logger := s.logger.WithValues("Namespace", s.ns, "Name", s.name)
+
+	name, err := s.Name()
+	if err != nil {
+		logger.Error(err, "Error creating unique secret name hash")
```

Comment:
This error wraps another error with the same description, please change it to reflect the right layer or information; for example "Creating secret" so a developer can co-relate error and intention.

Reply:
I have removed the previous description.

So, "Error creating unique secret name hash" is used only once
 -> https://github.com/redhat-developer/service-binding-operator/pull/750/files#diff-0141e5e3383464862fa2299425f3318875159d89221a6cd2f36088f6d06d75b1R71